provider/cpu_utils.py

```diff
@@ -98,8 +100,21 @@ def check_if_vm_vcpu_topology_match(session, os_type, cpuinfo):
                               " returned by guest:\n%s", out)
                 return False
 
+    threads_matched = cpuinfo.threads == threads
```

Comment:
Is this for linux only ?  we can move it  to the L70 block, what do you think about this ?